server/lib/manifest.py

```diff
@@ -0,0 +1,389 @@
+        self.manifest['createdBy'] = {
+            "@id": tale['authors'],
+            "@type": "schema:Person",
+            "schema:givenName": tale_user.get('firstName', ''),
+            "schema:familyName": tale_user.get('lastName', ''),
+            "schema:email": tale_user.get('email', '')
+        }
+
+    def create_context(self):
+        """
+        Creates the manifest namespace. When a new vocabulary is used, it shoud
+        get added here.
+        :return: A structure defining the used vocabularies
+        """
+        return {
+            "@context": [
+                "https://w3id.org/bundle/context",
+                {"schema": "http://schema.org/"}
+            ]
+        }
+
+    def create_dataset_record(self, user, folder_id):
+        """
+        Creates a record that describes a Dataset
+        :param user: The user
+        :param folder_id: Folder that represents a dataset
+        :return: Dictionary that describes a dataset
+        """
+        folder = self.folderModel.load(folder_id,
+                                       user=user,
+                                       force=True)
+        if folder:
+            meta = folder.get('meta')
+            if meta:
+                provider = meta.get('provider')
+                if provider:
+                    return {
+                        "@id": meta.get('identifier'),
+                        "@type": "Dataset",
+                        "name": folder['name'],
+                        "identifier": meta.get('identifier'),
+                        "publisher": self.publishers[provider]}
+
+    def create_aggregation_record(self, uri, bundle=None, parent_dataset=None):
+        """
+        Creates an aggregation record. Externally defined aggregations should include
+        a bundle and a parent_dataset if it belongs to one
+        :param uri:
+        :param bundle:
+        :param parent_dataset:
+        :return: Dictionary representing an aggregated file
+        """
+        aggregation = dict()
+        aggregation['uri'] = uri
+        if bundle:
+            aggregation['bundledAs'] = bundle
+        if parent_dataset:
+            aggregation['schema:isPartOf'] = parent_dataset
+        return aggregation
+
+    def add_item_records(self, user, tale):
+        """
+        Creates records for a set of item ids. This is desired when the mainfest is being generated
+        for a subset of files in a Tale. Note that these records get added to the internal manifest
+        object.
+        :param user: The user that is creating the manifest
+        :param tale: The tale whose information is being used
+        """
+        for item_id in self.item_ids:
+            item = self.itemModel.load(item_id,
+                                       user=user,
+                                       force=True)
+            if item:
+                root = self.itemModel.parentsToRoot(item, user=user)
```

Comment:
Since you're not interested in models of each path component, but rather an actual path:
```
from girder.utility import path as path_util
path_util.getResourcePath('item', item, user=user)
```
should save you the need of recreating it later on.